tensorflow/inference/docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -173,60 +186,121 @@ def _handle_load_model_post(self, res, data):  # noqa: C901
                 p = subprocess.Popen(cmd.split())
 
                 tfs_utils.wait_for_model(
-                    self._model_tfs_rest_port[model_name], model_name, self._tfs_wait_time_seconds
+                    rest_port, model_name, self._tfs_wait_time_seconds, p.pid
                 )
 
                 log.info("started tensorflow serving (pid: %d)", p.pid)
-                # update model name <-> tfs pid map
-                self._model_tfs_pid[model_name] = p
 
-                res.status = falcon.HTTP_200
-                res.body = json.dumps(
-                    {
-                        "success": "Successfully loaded model {}, "
-                        "listening on rest port {} "
-                        "and grpc port {}.".format(
-                            model_name,
-                            self._model_tfs_rest_port,
-                            self._model_tfs_grpc_port,
-                        )
-                    }
-                )
+                return {'status': falcon.HTTP_200,
+                        'body': json.dumps(
+                            {
+                                "success": "Successfully loaded model {}, "
+                                           "listening on rest port {} "
+                                           "and grpc port {}.".format(model_name, rest_port, grpc_port)
+                            },
+                        ),
+                        'pid': p.pid
+                        }
             except MultiModelException as multi_model_exception:
-                self._cleanup_config_file(tfs_config_file)
-                self._cleanup_config_file(batching_config_file)
                 if multi_model_exception.code == 409:
-                    res.status = falcon.HTTP_409
-                    res.body = multi_model_exception.msg
+                    return {'status': falcon.HTTP_409,
+                            'body': multi_model_exception.msg,
+                            'pid': multi_model_exception.pid, }
                 elif multi_model_exception.code == 408:
-                    res.status = falcon.HTTP_408
-                    res.body = multi_model_exception.msg
+                    cpu_memory_usage = tfs_utils.get_cpu_memory_util()
+                    log.info(f'cpu memory usage {cpu_memory_usage}')
+                    if cpu_memory_usage > 70:
```

Comment:
Is this requirement documented somewhere, where "70" is considered the right value?

Reply:
I did some experiment and try to load multiple large models to trigger OOM issue. The lowest memory threshold which triggers OOM is 73% of CPU memory usage. In normal use case, sagemaker will monitor memory usage and issue a load model request.